api/src/db/models/opportunity_models.py

```diff
@@ -49,72 +49,65 @@ class Opportunity(Base, TimestampMixin):
     publisher_user_id: Mapped[int | None]
     publisher_profile_id: Mapped[int | None]
 
-    summary: Mapped["OpportunitySummary | None"] = relationship(
-        back_populates="opportunity", single_parent=True, cascade="all, delete-orphan"
-    )
-
     opportunity_assistance_listings: Mapped[list["OpportunityAssistanceListing"]] = relationship(
         back_populates="opportunity", uselist=True, cascade="all, delete-orphan"
     )
-    link_funding_instruments: Mapped[list["LinkFundingInstrumentOpportunity"]] = relationship(
-        back_populates="opportunity", uselist=True, cascade="all, delete-orphan"
-    )
-    link_funding_categories: Mapped[list["LinkFundingCategoryOpportunity"]] = relationship(
-        back_populates="opportunity", uselist=True, cascade="all, delete-orphan"
-    )
-    link_applicant_types: Mapped[list["LinkApplicantTypeOpportunity"]] = relationship(
-        back_populates="opportunity", uselist=True, cascade="all, delete-orphan"
+
+    current_opportunity_summary: Mapped["CurrentOpportunitySummary | None"] = relationship(
+        back_populates="opportunity", single_parent=True, cascade="all, delete-orphan"
     )
 
     @property
-    def funding_instruments(self) -> list[FundingInstrument]:
-        # Helper method for serialization of the API response
-        return [f.funding_instrument for f in self.link_funding_instruments]
+    def summary(self) -> "OpportunitySummary | None":
+        """
+        Utility getter method for converting an Opportunity in our endpoints
 
-    @property
-    def funding_categories(self) -> list[FundingCategory]:
-        # Helper method for serialization of the API response
-        return [f.funding_category for f in self.link_funding_categories]
+        This handles mapping the current opportunity summary to the "summary" object
+         in our API responses - handling nullablity as well.
+        """
+        if self.current_opportunity_summary is None:
+            return None
 
-    @property
-    def applicant_types(self) -> list[ApplicantType]:
-        # Helper method for serialization of the API response
-        return [a.applicant_type for a in self.link_applicant_types]
+        return self.current_opportunity_summary.opportunity_summary
 
 
 class OpportunitySummary(Base, TimestampMixin):
     __tablename__ = "opportunity_summary"
 
-    opportunity_id: Mapped[int] = mapped_column(
-        ForeignKey(Opportunity.opportunity_id), primary_key=True
-    )
-    opportunity: Mapped[Opportunity] = relationship(Opportunity)
+    opportunity_summary_id: Mapped[int] = mapped_column(primary_key=True)
```

Comment:
At the moment, even when factoring in the history tables, nothing is more than a few hundred thousand records at the moment. I could make them big integers if we wanted to be very careful / not need to fix it later.

Reply:
Hmm, it might just be best to make all primary integer keys BigInteger, including on the opportunity table. I also noticed an incorrect column type on the Opportunity table - I might drop and remake that table as part of this PR as well for simplicity (the table isn't used yet) - will follow up with fixing that in the morning.